### ISSUE TITLE
gazebo only show black background

### ISSUE THREAD

**Issue by Lmaths** [osrf/docker_images#29]
I use parameter _**--device /dev/dri**_ to start the docker images osrf/ros:indigo-desktop-full. But When I run the command:
$  _**roslaunch  gazebo_ros  shapes_world.launch**_
The gazebo only show the black background.

But after remove the _**--device /dev/dri**_ parameter, the gazebo worked!
Any help will be appreciated, thanks!

**Comment by ruffsl** [comment]
What method are you using to enable the display, i.e. how are you going about connecting the Xserver for the display? Something from this [wiki](http://wiki.ros.org/docker/Tutorials/GUI) perhaps? Also, what hardware are you using?

**Comment by Lmaths** [comment]
@ruffsl 
Graphics card:  Intel Corporation Broadwell-U Integrated Graphics (rev 09)
Driver: i915

The method to enable the display:
$  **_xhost   +_**
$  **_docker   run   --device   /dev/dri      --security-opt="label:disable"   \
     --env="DISPLAY" --env QT_X11_NO_MITSHM=1  \ 
     --volume="/tmp/.X11-unix:/tmp/.X11-unix:rw"    \
      osrf/ros:indigo-desktop-full     rqt_**

And I installed the intel driver in the container following the [hardware acceleration wiki](http://wiki.ros.org/docker/Tutorials/Hardware%20Acceleration).

**Comment by ruffsl** [comment]
I don't have a PC with Intel Integrated Graphics nearby to test with, but from checking the wiki history, looks like Matt Vollrath has been contributing to the Intel section science after I started the wiki page. @mvollrath any idea of what this above amount to? @Lmaths did you 
> note that users in the container (other than root) need access to the 'video' group for Mesa DRI devices

**Comment by Lmaths** [comment]
I used the mesa-utils to test the video acceleration, and it ran with no errors.
`$  glxgears`
Only gazebo did not work correctly.

**Comment by Lmaths** [comment]
I reinstalled the intel i915 driver in the container using [a ppa respository](http://sourcedigit.com/17038-install-updated-optimized-graphics-drivers-on-ubuntu-15-04-ubuntu-14-04/) that solved this problem.

Details:
`# echo "deb http://ppa.launchpad.net/oibaf/graphics-drivers/ubuntu trusty main" >  /etc/apt/sources.list.d/intel-video-driver.list  `
` # apt-key adv --keyserver keyserver.ubuntu.com --recv-keys 957D2708A03A4626 `
` # apt-get update `
` # apt-get install xserver-xorg-video-intel `

**Comment by bnederkorn** [comment]
I had the issue that the gazebo simulation (version 7.5.0), which I started inside a ros Humble docker container, showed only a black GUI. First, I tried to run the simulation on my PC, and it worked fine. So I looked inside the container with `docker exec -it containerID /bin/bash` and tried `apt update`. I relised that there were over 300 packages that can be upgraded. So I upgraded them, stoped the container and restarted it. And now it works fine.
